accounting/invoice/models.py

```diff
@@ -20,3 +20,376 @@
+    """
+    return timezone.now() + timedelta(days=offset)
+
+
+class Invoice(UuidModel, TimeStampedModel):
+    """
+    A model to hold all data related to an invoice.
+
+    Can be used to generate actual HTML/PDF invoices off of templates.
+    """
+
+    PDF_OPTIONS = {
+        'page-size': 'A4',
+        'margin-top': '10mm',
+        'margin-right': '0mm',
+        'margin-bottom': '10mm',
+        'margin-left': '0mm',
+        'encoding': "UTF-8",
+    }
+
+    number = models.CharField(
+        max_length=80, default=utils.default_invoice_number,
+        help_text=_("The unique invoice number. "
+                    "Defaults to yyyy-mm for the current year and month."))
+    date = models.DateTimeField(
+        default=timezone.now,
+        help_text=_("The date this invoice was created and sent for billing purposes. "
+                    "Defaults to right now, but can be changed."))
+    billing_start_date = models.DateTimeField(
+        default=utils.get_first_day_past_month,
+        help_text=_("The first date for which the line items in this invoice were provided. "
+                    "Defaults to the first day of the past month."))
+    billing_end_date = models.DateTimeField(
+        default=utils.get_last_day_past_month,
+        help_text=_("The last date for which the line items in this invoice were provided. "
+                    "Defaults to the last day of the past month."))
+    due_date = models.DateTimeField(
+        default=get_day_with_offset,
+        help_text=_("When this invoice should be paid."))
+    provider = models.ForeignKey(
+        Account, on_delete=models.CASCADE, related_name='provider_invoices',
+        help_text=_("The invoicing service/product provider."))
+    client = models.ForeignKey(
+        Account, on_delete=models.CASCADE, related_name='client_invoices',
+        help_text=_("The client being invoiced for services/products."))
+    paid = models.BooleanField(
+        default=False,
+        help_text=_("Whether this invoice has been paid by the client yet or not."))
+    extra_text = models.TextField(
+        blank=True, null=True,
+        help_text=_("Any arbitrary extra text that the provider would like to display on their invoice. "
+                    "Each template has a designated location to place this extra text."))
+    signature = models.ImageField(
+        blank=True, null=True,
+        help_text=_("The provider's signature."))
+    template = models.CharField(
+        max_length=80, choices=InvoiceTemplate.choices, default=InvoiceTemplate.Default,
+        help_text=_("The template to use to generate this invoice."))
+    history = HistoricalRecords()
+
+    # TODO: The following 2 fields are temporary while we don't have a UI besides the Django admin.
+    # TODO: The front-end will eventually call endpoints to do what these fields allow us to do.
+    auto_download_jira_worklogs_on_save = models.BooleanField(  # pylint: disable=invalid-name
```

Comment:
I'm not sure of the functionality yet, that's something to do discovery on. But most likely, it will remain like this until we can get a UI. Then from the UI, the user's worklogs are automatically refreshable, and are initially created whenever a monthly job is run to email the team members that the invoice cycle has begun.